qml/MainContent.qml

```diff
@@ -86,25 +86,70 @@ Item {
         }
         GridLayout {
             Layout.fillWidth: true
-            columns: 2
+            columns: 3
 
             Label {
                 text: qsTr("Next break:")
             }
+
             TimeProgressBar {
+                id: nextBreakTimeProgressBar
                 Layout.fillWidth: true
 
                 maxValue: controller.settings.breakInterval
                 value: controller.timer.elapsedWorkPeriod
+
+                onTimeValueChanged:
+                {
+                    var timeDiff = newValue - controller.timer.elapsedWorkPeriod
+                    controller.timer.elapsedWorkPeriod = newValue
+                    controller.timer.elapsedWorkTime += timeDiff
+                }
+            }
+
+            ImageButton {
+                enabled: !nextBreakTimeProgressBar.timeEditionInProgress
+                styleFont: Style.font.imageButtonSmallest
+                type: "edit"
+                tooltip: qsTr("Edit next break")
+
+                onClicked:
+                {
+                    nextBreakTimeProgressBar.startTimeEdition()
+                }
             }
+
             Label {
                 text: qsTr("Work time:")
             }
+
             TimeProgressBar {
+                id: workTimeProgressBar
                 Layout.fillWidth: true
 
                 maxValue: controller.settings.workTime
                 value: controller.timer.elapsedWorkTime
+
+                onTimeValueChanged:
+                {
+                    var timeDiff = newValue - controller.timer.elapsedWorkTime
+                    controller.timer.elapsedWorkTime = newValue
+
+                    var newElapsedWorkTime = controller.timer.elapsedWorkPeriod + timeDiff
```

Comment:
I agree that we should update `elapsedWorkTime` when we edit `elapsedWorkPeriod` but I disagree the other way.

For example if I realize after 1h of work, that I forgot to start Resto, so I start and continue my work. Some time later I edit Work Time to add the 1h - but it doesn't necessary mean that I should immediately take a break now.

Reply:
Done